emsdk

```diff
@@ -75,10 +75,19 @@ if platform.mac_ver()[0] != '':
   ENVPATH_SEPARATOR = ':'
 
 LINUX = False
+LINUX_ARM = False
```

Comment:
Perhaps instead of having a variable `LINUX_ARM`, we could have a variable `ARCH` that is one of `'x86', `'x64'`, '`arm'` or `'aarch64'`? that would be more future proof to avoid growing more variables. (In future a refactor might change OSX/WINDOWS/LINUX vars to an `OS='win'/'linux'/'mac'` string)

Then to abbreviate checks, could have helpers
```py
def arch_is_arm():
  return ARCH == 'arm' or ARCH == 'aarch64'
```
that way the amount of state will stay minimal/nonredudant, and e.g. it wouldn't be possible to mismatch `LINUX_ARM=True` with `LINUX=False`.

Reply:
Yeah that'd be better in the long run, I'll try that this weekend.